fakebook/src/app/component/profile-view/profile-view.component.ts

```diff
@@ -41,22 +41,20 @@ export class ProfileViewComponent implements OnInit {
   }
 
   async getUser(): Promise<void> {
-    let tempEmail = '';
-    // Async problems possibly
-    this.profileService.GetProfile(this.currentUserEmail).subscribe(profile => this.selfUser = profile);
 
     if (this.route.snapshot.paramMap.get('email') != null) {
-      tempEmail += this.route.snapshot.paramMap.get('email');
 
-      const email = tempEmail;
-
-      this.profileService.GetProfile(email).subscribe(user => this.user = user);
-
-      this.postService.getUserPosts(email).subscribe(posts => this.posts = posts);
-      // This might have problems due to async. currentUserEmail may not be set in time
-      this.profileService.GetProfile(email)
+      const email = this.route.snapshot.paramMap.get('email');
+      if (email) {
+        // Set user
+        this.profileService.GetProfile(email).subscribe(user => this.user = user);
+        // Set posts
+        this.postService.getUserPosts(email).subscribe(posts => this.posts = posts);
+        // Set follow status
+        this.profileService.GetProfile(email)
         .subscribe(user => this.profileService.GetProfile(this.currentUserEmail)
           .subscribe(selfUser => this.followStatus = this.followService.getFollowStatus(selfUser, user)));
```

Comment:
this isn't following rxjs good practices but if it works it works

Reply:
I will research this but for now we are also thinking if it works it works